ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/SparseTransactions.java

```diff
@@ -273,6 +273,16 @@ public OptionalLong getNextNonceFor(final Address sender) {
     return OptionalLong.empty();
   }
 
+  @Override
+  public OptionalLong getCurrentNonceFor(final Address sender) {
+    final var senderTxs = txsBySender.get(sender);
+    if (senderTxs != null) {
+      final var gap = gapBySender.get(sender);
```

Comment:
This is on purpose, since the data structures must be always aligned on consistent, `txsBySender` is the reference one, and if a tx is there, the it must be also in any other related data structure, and so any inconsistency should triggers an error, and fail fast, because it means there is a bug, and that helped to spot many edge cases during the development.